bolt-http4k/src/main/java/com/slack/api/bolt/http4k/Http4kSlackApp.java

```diff
@@ -19,32 +20,94 @@
 import static java.util.stream.Collectors.toMap;
 import static org.http4k.core.ParametersKt.toParameters;
 import static org.http4k.core.Status.INTERNAL_SERVER_ERROR;
+import static org.http4k.core.Status.NOT_FOUND;
 
 public class Http4kSlackApp implements Function1<Request, Response> {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(Http4kSlackApp.class);
+    protected static final Logger LOGGER = LoggerFactory.getLogger(Http4kSlackApp.class);
 
-    private final App app;
-    private final SlackRequestParser requestParser;
+    protected final App app;
+    protected final String path;
+    protected final SlackRequestParser requestParser;
 
     public Http4kSlackApp(App app) {
+        this(app, "/slack/events");
+    }
+
+    public Http4kSlackApp(App app, String path) {
         this.app = app;
+        this.path = path;
         requestParser = new SlackRequestParser(app.config());
     }
 
     @Override
     public Response invoke(Request request) {
+        Method requestMethod = request.getMethod();
+        String requestPath = request.getUri().getPath();
+        if (requestMethod == null) {
+            // checking this, just in case
+            return notFound(request);
+        }
+        if (requestMethod == Method.POST && requestPath.equals(getSlackEventsRequestURI())) {
+            return handleEventRequest(request);
+        } else if (requestMethod == Method.GET) {
+            if (app.config().isOAuthInstallPathEnabled()
+                    && requestPath.equals(app.config().getOauthInstallRequestURI())) {
+                return handleOAuthRequest(request);
+            } else if (app.config().isOAuthInstallPathEnabled()
+                    && requestPath.equals(app.config().getOauthRedirectUriRequestURI())) {
+                return handleOAuthRequest(request);
+            }
+        }
+        return notFound(request);
+    }
+
+    // ----------------------------------------------------------
+
+    protected String getSlackEventsRequestURI() {
```

Comment:
I changed all internal methods' access modifiers from `private` to `protected` so as to enable developers override them in their sub class as necessary.